### ISSUE TITLE
MagicOnion.Generated.cs fails to compile in unity

### ISSUE THREAD

**Issue by wmqKevin** [Cysharp/MagicOnion#702]
Every time I declare a function with multiple parameters, an error is reported below:
![image](https://github.com/Cysharp/MagicOnion/assets/33646381/0dcd1d2e-e8aa-4d17-8dfa-391446a64be8)
 FormatterNotRegisteredException: MagicOnion.DynamicArgumentTuple`2[[Shared.MessagePackObjects.JoinRequest, Assembly-CSharp, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null],[Shared.MessagePackObjects.TransformRequest, Assembly-CSharp, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null]] is not registered in resolver. 
I don't know is that I didn't generate MagicOnion.Generated.cs. But the MagicOnion.Generated.cs I generated through MagicOnion.MSBuild.Tasks fails to compile every time. Like these images
![image](https://github.com/Cysharp/MagicOnion/assets/33646381/366c0b2d-1ca3-43be-bf80-5f08a9039b44)
![image](https://github.com/Cysharp/MagicOnion/assets/33646381/7395fda9-4097-450c-8caf-5bb7efc5c2fa)

How can I solve this problem?

**Comment by mayuki** [comment]
MSBuild.Task is no longer supported in MagicOnion 5.x.
MessagePack Formatter error may be due to the Resolver not registered.

https://github.com/Cysharp/MagicOnion/blob/release/5.x/README.md#messagepack-for-c-1

**Comment by wmqKevin** [comment]
> MSBuild.Task is no longer supported in MagicOnion 5.x. MessagePack Formatter error may be due to the Resolver not registered.
> 
> https://github.com/Cysharp/MagicOnion/blob/release/5.x/README.md#messagepack-for-c-1

Thank you @mayuki 

the problem was solved after I manually added it. However, when I declare a function, if there are multiple parameters, MagicOnion will generate the corresponding DynamicArgumentTuple, but I haven't found a way to automatically generate the corresponding Formatter. Can it be generated automatically?

**Comment by mayuki** [comment]
If code is generated by generator, it will automatically generate the corresponding Formatter for the argument.
https://github.com/Cysharp/MagicOnion/blob/release/5.x/samples/ChatApp/ChatApp.Unity/Assets/Scripts/Generated/MagicOnion.Generated.cs#L148

**Comment by wmqKevin** [comment]
> If code is generated by generator, it will automatically generate the corresponding Formatter for the argument. https://github.com/Cysharp/MagicOnion/blob/release/5.x/samples/ChatApp/ChatApp.Unity/Assets/Scripts/Generated/MagicOnion.Generated.cs#L148

Thank you @mayuki 
I looked at the content of "**ChatApp.Shared.csproj**" and solved the problem, but a new problem has emerged - -.
When my Server serializes UnityEngine.Vector3, a similar **FormatterNotRegisteredException** error occurs. I tried to register it in the Resolver using the **MagicOnionResolver** in Unity, but it seems to have no effect.
![image](https://github.com/Cysharp/MagicOnion/assets/33646381/c181852f-6496-4fd0-b713-06b169a44a8c)
Is there any way to register Unity's data structures on the server side, or do I need to define new structures myself?
